File: CodeArchieve/UOJ/Test-Round-1/83-水题出题人/problem_83/counting_sort.cpp
#include <stdio.h>

int counter, n, max;
int a[1000001];
int b[10000001];

void counting_sort_main(){
	int i, j;
	counter++;
	scanf("%d", &n);
	counter++;
	max = 0;
	counter++;
	for(i = 1; i <= n; i++){
		counter++;
		scanf("%d", &a[i]);
		counter++;
		if(a[i] > max){
			counter++;
			max = a[i];
		}
	}
	counter++;
	for(i = 0; i <= max; i++){
		counter++;
		b[i] = 0;
	}
	counter++;
	for(i = 1; i <= n; i++){
		counter++;
		b[a[i]]++;
	}
	counter++;
	for(i = 0; i <= max; i++){
		counter++;
		for(j = 1; j <= b[i]; j++){
			counter++;
//			printf("%d ", i);
		}
	}
}

int main(){
	counter = 0;
	counting_sort_main();
	printf("\n");
	if(counter <= 2000000){
		printf("Accepted! The counter value is :%d\n", counter);
		return 0;
	}else{
		printf("Time Limit Exceeded because the counter value is :%d\n", counter);
		return 1;
	}
}

File: CodeArchieve/UOJ/Test-Round-1/83-水题出题人/problem_83/counting_sort.pas
program counting_sort;

var counter, n, max :longint;
var a :array[1..1000000] of longint;
var b :array[0..10000000] of longint;

procedure counting_sort_main;
var i, j :longint;
begin
	inc(counter);
	readln(n);
	inc(counter);
	max := 0;
	inc(counter);
	for i := 1 to n do begin
		inc(counter);
		readln(a[i]);
		inc(counter);
		if a[i] > max then begin
			inc(counter);
			max := a[i];
		end;
	end;
	inc(counter);
	for i := 0 to max do begin
		inc(counter);
		b[i] := 0;
	end;
	inc(counter);
	for i := 1 to n do begin
		inc(counter);
		inc(b[a[i]]);
	end;
	inc(counter);
	for i := 0 to max do begin
		inc(counter);
		for j := 1 to b[i] do begin
			inc(counter);
			write(i, ' ');
		end;
	end;
end;

begin
	counter := 0;
	counting_sort_main;
	writeln;
	if counter <= 2000000 then begin
		writeln('Accepted! The counter value is :', counter);
	end else begin
		writeln('Time Limit Exceeded because the counter value is :', counter);
	end;
end.
